pkg/util/glog/glog.go

```diff
@@ -1,101 +1,224 @@
 
-func (discard) Is(level int) bool                { return false }
-func (discard) V(level int) Logger               { return None }
-func (discard) Infof(_ string, _ ...interface{}) {}
+func (discard) Is(level int32) bool                    { return false }
+func (discarding discard) V(level int32) VerboseLogger { return discarding }
+func (discard) Infof(_ string, _ ...interface{})       {}
+func (discard) Info(_ ...interface{})                  {}
+func (discard) Errorf(_ string, _ ...interface{})      {}
+func (discard) Error(_ ...interface{})                 {}
+func (discard) Warningf(_ string, _ ...interface{})    {}
+func (discard) Warning(_ ...interface{})               {}
+func (discard) Fatalf(_ string, _ ...interface{})      {}
+func (discard) Fatal(_ ...interface{})                 {}
 
 // glogger outputs log messages to glog
 type glogger struct{}
 
-func (glogger) Is(level int) bool {
-	return bool(glog.V(glog.Level(level)))
+func (glogger) Is(level int32) bool {
+	return bool(log.V(log.Level(level)))
 }
 
-func (glogger) V(level int) Logger {
-	return gverbose{glog.V(glog.Level(level))}
+func (glogger) V(level int32) VerboseLogger {
+	return gverbose{log.V(log.Level(level))}
 }
 
 func (glogger) Infof(format string, args ...interface{}) {
-	glog.Infof(format, args...)
+	log.Infof(format, args...)
 }
 
-// gverbose handles glog.V(x) calls
-type gverbose struct {
-	glog.Verbose
+func (glogger) Info(args ...interface{}) {
+	log.Info(args...)
 }
 
-func (gverbose) Is(level int) bool {
-	return bool(glog.V(glog.Level(level)))
+func (glogger) Warningf(format string, args ...interface{}) {
+	log.Warningf(format, args...)
 }
 
-func (gverbose) V(level int) Logger {
-	if glog.V(glog.Level(level)) {
-		return Log
-	}
-	return None
+func (glogger) Warning(args ...interface{}) {
+	log.Warning(args...)
+}
+
+func (glogger) Errorf(format string, args ...interface{}) {
+	log.Errorf(format, args...)
+}
+
+func (glogger) Error(args ...interface{}) {
+	log.Error(args...)
+}
+
+func (glogger) Fatalf(format string, args ...interface{}) {
+	log.Fatalf(format, args...)
+}
+
+func (glogger) Fatal(args ...interface{}) {
+	log.Fatal(args...)
+}
+
+// gverbose handles glog.V(x) calls
+type gverbose struct {
+	log.Verbose
 }
 
 func (g gverbose) Infof(format string, args ...interface{}) {
 	g.Verbose.Infof(format, args...)
 }
 
-// file logs the provided messages at level or below to the writer, or delegates
+func (g gverbose) Info(args ...interface{}) {
+	g.Verbose.Info(args...)
+}
+
+// FileLogger logs the provided messages at level or below to the writer, or delegates
 // to glog.
-type file struct {
-	w     io.Writer
-	level int
+type FileLogger struct {
+	w     *bufio.Writer
+	level int32
 }
 
-func (f file) Is(level int) bool {
-	return level <= f.level
+// Is returns whether the current logging level is greater than or equal to the parameter
+func (f *FileLogger) Is(level int32) bool {
+	return level <= atomic.LoadInt32(&f.level)
```

Comment:
There are already go routines using the logging library. Without having researched go's memory consistency contracts, I can't promise the outcome will be deterministic. That said, I will remove the guard.
If the desired outcome is the fixed > 2 implementation, this issue is moot since the value will no longer be dynamic.